x/gov/msg.go

```diff
@@ -20,6 +22,23 @@ func (CreateTextProposalMsg) Path() string {
 }
 
 func (m CreateTextProposalMsg) Validate() error {
+	err := m.Author.Validate()
+	switch {
+	case len(m.ElectorateID) == 0:
+		return errors.Wrap(errors.ErrInvalidInput, "empty electorate id")
+	case len(m.ElectionRuleID) == 0:
+		return errors.Wrap(errors.ErrInvalidInput, "empty election rules id")
+	case m.StartTime.Time().IsZero():
```

Comment:
This is an awkward case. `UnixTime` has [`Validate`](https://github.com/iov-one/weave/blob/master/time.go#L79) method that you can use. It has a similar time range to `time.Time` but allows zero UNIX time value. Might be that the only proper way to test it is to use a pointer. That would cause an ugly API though.

I guess the issue here is that you want to explicit declaration of time and if not given it is initialized with a zero value. If so, then you can just check `case m.StartTime == 0:`

Reply:
Thanks, I have added the `startTime == 0` case which makes sense for me but it does not cover the https://github.com/iov-one/weave/pull/500/files#diff-84a65f4bb1934b67522bd91707d73beaR157 test.